### ISSUE TITLE
Handle image submits

### ISSUE THREAD

**Issue by mjpieters** [MechanicalSoup/MechanicalSoup#201]
An `input` element of [type `image`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/image) is also a valid submit button:

```
<form><input type="text" name="search" /><input type="image" name="search" /></form>
```

Clicking on the image button will submit the form, as does hitting enter. The form then has two fields, `image.x` and `image.y`, set to the coordinates where the image was clicked, or `0` when hitting enter.

MechanicalSoup has no facilities for handling this type of input; it is not recognised as a valid submit and on submitting, no `.x` and `.y` entries are generated.

**Comment by moy** [comment]
Indeed. Supporting `image` as a synonym for `submit` is straightforward. Support for `.x` and `.y` fields needs a bit of thinking: what should the MechanicalSoup API look like?

Perhaps

    br["image.x"] = x
    br["image.y"] = y
    br.submit_selected()

?

Then `br["image.x"]` would just check that there's an image input with name `image`, and store somewhere the value `x`. The easy implementation would be to add a hidden field to store the value. Perhaps it'd be nicer to store the value as part of the `image` tag, adding fields like `data-x` and `data-y`?
